### ISSUE TITLE
[C++][Python][R] PrettyPrint ignores timezone

### ISSUE THREAD

**Issue by asfimport** [apache/arrow#30117]
When printing TimestampArray in pyarrow the timezone information is ignored by PrettyPrint (__str__  calls to_string() in array.pxi).

```python

import pyarrow as pa

a = pa.array([0], pa.timestamp('s', tz='+02:00'))

print(a) # representation not correct?
# <pyarrow.lib.TimestampArray object at 0x7f834c7cb9a8>
# [
#  1970-01-01 00:00:00
# ]
```

**Reporter**: [Alenka Frim](https://issues.apache.org/jira/browse/ARROW-14567) / @AlenkaF
**Watchers**: [Rok Mihevc](https://issues.apache.org/jira/browse/ARROW-14567) / @rok
#### Related issues:
- [PrettyPrint Improvements](https://github.com/apache/arrow/issues/33527) (is a child of)
- [[Python] datetime shifted when using pyarrow.Table.from_pandas to load a pandas DateFrame containing datetime with timezone](https://github.com/apache/arrow/issues/20493) (causes)
- [[R] fix behaviour when converting timestamps with "" as tzone](https://github.com/apache/arrow/issues/30005) (relates to)

<sub>**Note**: *This issue was originally created as [ARROW-14567](https://issues.apache.org/jira/browse/ARROW-14567). Please see the [migration documentation](https://github.com/apache/arrow/issues/14542) for further details.*</sub>

**Comment by AlenkaF** [comment]
I created a PR for the simple version of the fix which is adding a "Z" to the end of the string in case timezone is defined.

```python
>>> import pyarrow as pa

>>> # tz defined
>>> pa.array([0], pa.timestamp('s', tz='+02:00'))
<pyarrow.lib.TimestampArray object at 0x131399fc0>
[
  1970-01-01 00:00:00Z
]

>>> # tz not defined
>>> pa.array([0], pa.timestamp('s'))
<pyarrow.lib.TimestampArray object at 0x13139a020>
[
  1970-01-01 00:00:00
]
```

I haven't added any change in the type output as the timezone information is already included:

```python
>>> a = pa.array([0], pa.timestamp('s', tz='+02:00'))
>>> a.type
TimestampType(timestamp[s, tz=+02:00])
```

or do we want info about the timezone in array object print?

**Comment by rok** [comment]
Thanks for doing this @AlenkaF !
Adding Z is definitely an improvement and removes current ambiguity.

Did you consider adding an option to print in local time as per [ISO8601 convention](https://en.wikipedia.org/wiki/ISO_8601) displaying local time + offset from UTC?
This could be done with logic from [local_timestamp](https://arrow.apache.org/docs/cpp/compute.html#timezone-handling), what's missing is logic to add timezone offset information. This is currently won't work for cases where timezone is given as a fixed offset (see #12865) but those cases could be handled as exceptions. Given that we could also make a separate issue.

**Comment by AlenkaF** [comment]
Printing in local time would definitely be good to have. What I was thinking is to have the simple fix asap and then do a follow-up with localized strings when possible.
